src/read/pipelining.rs

```diff
@@ -0,0 +1,1209 @@
+                    .map(|(path, perms)| (path, perms.mode() & 0o777))
+                    .collect::<Vec<_>>(),
+                vec![
+                    (td.path().join("a/b/c"), 0o765),
+                    (td.path().join("a/b"), 0o500),
+                ]
+            );
+        }
+    }
+}
+
+#[cfg(unix)]
+pub mod split_extraction {
+    use displaydoc::Display;
+    use thiserror::Error;
+
+    use std::fs;
+    use std::io;
+    use std::mem::{self, MaybeUninit};
+    use std::path::Path;
+    use std::sync::mpsc;
+    use std::thread;
+
+    use crate::compression::CompressionMethod;
+    use crate::read::ZipArchive;
+    use crate::read::{make_crypto_reader, make_reader};
+    use crate::result::ZipError;
+    use crate::spec::FixedSizeBlock;
+    use crate::types::{ZipFileData, ZipLocalEntryBlock};
+
+    #[cfg(not(target_os = "linux"))]
+    use crate::read::split::pipe::unix::{PipeReadBufferSplicer, PipeWriteBufferSplicer};
+    #[cfg(target_os = "linux")]
+    use crate::read::split::{
+        file::linux::FileCopy,
+        pipe::linux::{PipeReadSplicer, PipeWriteSplicer},
+    };
+    use crate::read::split::{
+        file::{
+            unix::{FileBufferCopy, FileInput, FileOutput},
+            CopyRange, InputFile,
+        },
+        pipe::{unix::create_pipe, ReadSplicer, WriteSplicer},
+        util::{copy_via_buf, TakeWrite},
+    };
+
+    use super::{
+        handle_creation::{
+            transform_entries_to_allocated_handles, AllocatedHandles, HandleCreationError,
+            ZipDataHandle,
+        },
+        path_splitting::{lexicographic_entry_trie, PathSplitError},
+    };
+
+    /// Errors encountered during the split pipelined extraction process.
+    #[derive(Debug, Display, Error)]
+    pub enum SplitExtractionError {
+        /// i/o error: {0}
+        Io(#[from] io::Error),
+        /// zip error: {0}
+        Zip(#[from] ZipError),
+        /// path split error: {0}
+        PathSplit(String),
+        /// handle creation error: {0}
+        HandleCreation(#[from] HandleCreationError),
+    }
+
+    impl<'a> From<PathSplitError<'a>> for SplitExtractionError {
+        fn from(e: PathSplitError<'a>) -> Self {
+            let msg = format!("{}", e);
+            Self::PathSplit(msg)
+        }
+    }
+
+    /* TODO: make this share code with find_data_start()! */
+    fn get_or_find_data_start<InF>(data: &ZipFileData, input_file: InF) -> Result<u64, ZipError>
+    where
+        InF: InputFile,
+    {
+        // TODO: use .get_or_try_init() once stabilized to provide a closure returning a Result!
+        if let Some(data_start) = data.data_start.get() {
+            return Ok(*data_start);
+        }
+
+        let block = {
+            let block: MaybeUninit<[u8; mem::size_of::<ZipLocalEntryBlock>()]> =
+                MaybeUninit::uninit();
+            let mut block: [MaybeUninit<u8>; mem::size_of::<ZipLocalEntryBlock>()] =
+                unsafe { mem::transmute(block) };
```

Comment:
Use `pod_read_unaligned` to encapsulate the unsafe code. Also, call `from_le()` to convert the endianness if necessary.